framework/sandstone.cpp

```diff
@@ -3762,7 +3758,10 @@ int main(int argc, char **argv)
             }
         }
 
-        lastTestResult = run_one_test(*it, per_cpu_failures);
+        if (it->test->quality_level >= sApp->requested_quality)
```

Comment:
This check does not belong in `main()`. It belongs to where the check was originally performed, inside `get_(first
|next)_test()` functions. You can opt to reorganize the code so that it checks and outputs there. Maybe it is even worth unifying the two functions.